src/m365/teams/commands/app/app-list.ts

```diff
@@ -47,100 +42,38 @@ class TeamsAppListCommand extends GraphCommand {
   #initTelemetry(): void {
     this.telemetry.push((args: CommandArgs) => {
       Object.assign(this.telemetryProperties, {
-        all: args.options.all || false,
-        teamId: typeof args.options.teamId !== 'undefined',
-        teamName: typeof args.options.teamName !== 'undefined'
+        distributionMethod: typeof args.options.distributionMethod !== 'undefined'
       });
     });
   }
 
   #initOptions(): void {
     this.options.unshift(
       {
-        option: '-a, --all'
-      },
-      {
-        option: '-i, --teamId [teamId]'
-      },
-      {
-        option: '-t --teamName [teamName]'
+        option: '--distributionMethod',
+        autocomplete: TeamsAppListCommand.allowedDistributionMethods
       }
     );
   }
 
   #initValidators(): void {
     this.validators.push(
       async (args: CommandArgs) => {
-        if (args.options.teamId && args.options.teamName) {
-          return 'Specify either teamId or teamName, but not both.';
-        }
-
-        if (args.options.teamId && !validation.isValidGuid(args.options.teamId)) {
-          return `${args.options.teamId} is not a valid GUID`;
+        if (args.options.distributionMethod &&
+          TeamsAppListCommand.allowedDistributionMethods.indexOf(args.options.distributionMethod) < 0) {
+          return `'${args.options.distributionMethod}' is not a valid distribution method. Allowed distribution methods are ${TeamsAppListCommand.allowedDistributionMethods.join(', ')}`;
```

Comment:
Let's specify the option name in the error to make it more clear what option is wrong.

```suggestion
          return `'${args.options.distributionMethod}' is not a valid distributionMethod. Allowed distribution methods are ${TeamsAppListCommand.allowedDistributionMethods.join(', ')}`;
```